File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int ans=-1;
        for(int j=0; j<haystack.length();j++) {
            if(needle[0]==haystack[j]) {
                ans=j;
                for(int i=1;i<needle.length();i++){
                    if(needle[i]!=haystack[j+i]) {
                        cout<<ans<<" ";
                        ans=-1;
                        break;
                    }
                    else continue;
                    
                }
                if(ans!=-1) return ans;
            }
            
        }
    return ans;}
    
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        for i in range(len(haystack)-len(needle)+1):
            if haystack[i]==needle[0]:
                k=i
                cnt=0
                for j in range(len(needle)):
                    if haystack[k]==needle[j]:
                        cnt+=1
                        k+=1
                    else : break
                if cnt==len(needle): return i
        return -1
                
        